src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/auth/adal_authenticator.py

```diff
@@ -0,0 +1,45 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import adal
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class AdalAuthenticator:
```

Comment:
@yugangw-msft for auth-related code.

Reply:
FYI, This code was already in the core CLI, I just refactored it into an external class because it was mingled with the bot creation code.

The purpose of this is, to create a bot we need a Microsoft Application Id. Users can either a) provide an id or b) ask us to provision one for them, case in which we authenticate to request permissions to provision an application and we create it and use it for the bot.

But again this was in custom.py before this PR so you already reviewed it likely.

Happy to provide more details!